src/main/scala/org/graphframes/GraphFrame.scala

```diff
@@ -265,6 +266,85 @@ class GraphFrame private(
     edges.select(explode(array(SRC, DST)).as(ID)).groupBy(ID).agg(count("*").cast("int").as("degree"))
   }
 
+  // ========================= Partition By ====================================
+  val PARTITION_ID: String = "partition_id"
+
+  /**
+   * A [[org.apache.spark.Partitioner]] that use the key of PairRDD as partition
+   * id number.
+   */
+  class ExactAsKeyPartitioner(partitions: Int) extends Partitioner {
+    require(partitions >= 0, s"Number of partitions ($partitions) cannot be negative.")
+
+    def numPartitions: Int = partitions
+
+    def getPartition(key: Any): Int = {
+      val partitionIdAsKey = key.asInstanceOf[Int]
+      return partitionIdAsKey
+    }
+  }
+
+  /**
+   * Implements a watered down version of Graphx partitionBy.
+   *
+   * @param numPartitions Number of partitions to be split by
+   * @param strategy any case object of Graphx's PartitionStrategy trait
```

Comment:
perhaps update the text to match http://spark.apache.org/docs/latest/api/scala/index.html#org.apache.spark.graphx.Graph@partitionBy(partitionStrategy:org.apache.spark.graphx.PartitionStrategy,numPartitions:Int):org.apache.spark.graphx.Graph[VD,ED]

```
Repartitions the edges in the graph according to partitionStrategy.
partitionStrategy
the partitioning strategy to use when partitioning the edges in the graph.
numPartitions
the number of edge partitions in the new graph.
```